frame/transaction-payment/src/lib.rs

```diff
@@ -271,9 +271,8 @@ decl_module! {
 			// that if we collapse to minimum, the trend will be positive with a weight value
 			// which is 1% more than the target.
 			let min_value = T::FeeMultiplierUpdate::min();
-			let mut target =
-				T::FeeMultiplierUpdate::target() *
-				(T::AvailableBlockRatio::get() * T::MaximumBlockWeight::get());
+			let mut target = T::FeeMultiplierUpdate::target() *
+				T::block_weights().get(DispatchClass::Normal).max_total.unwrap();
```

Comment:
this unwrap might require a message that transaction-payment doesn't allow a unlimited max_total for Normal dispatchable maybe ?